Sources/BraintreeCore/Analytics/BTAnalyticsService.swift

```diff
@@ -38,53 +16,53 @@ class BTAnalyticsService: Equatable {
-
-    ///  Tracks an event.
-    ///
-    ///  Events are queued and sent in batches to the analytics service, based on the status of the app and
-    ///  the number of queued events. After exiting this method, there is no guarantee that the event has been sent.
-    /// - Parameter eventName: String representing the event name
+    
+    /// Sends analytics event to https://api.paypal.com/v1/tracking/batch/events/ via a background task.
+    /// - Parameters:
+    ///   - eventName: Name of analytic event.
+    ///   - errorDescription: Optional. Full error description returned to merchant.
+    ///   - correlationID: Optional. CorrelationID associated with the checkout session.
+    ///   - payPalContextID: Optional. PayPal Context ID associated with the checkout session.
     func sendAnalyticsEvent(
         _ eventName: String,
         errorDescription: String? = nil,
         correlationID: String? = nil,
         payPalContextID: String? = nil
     ) {
-        DispatchQueue.main.async {
-            self.payPalContextID = payPalContextID
-            self.enqueueEvent(eventName, errorDescription: errorDescription, correlationID: correlationID)
-            self.flushIfAtThreshold()
+        Task(priority: .background) {
+            await performEventRequest(
+                eventName,
+                errorDescription: errorDescription,
+                correlationID: correlationID,
+                payPalContextID: payPalContextID
+            )
         }
     }
-
-    /// Sends request to FPTI immediately, without checking number of events in queue against flush threshold
-    func sendAnalyticsEvent(
+    
+    /// Exposed to be able to execute this function synchronously in unit tests
+    func performEventRequest(
         _ eventName: String,
         errorDescription: String? = nil,
         correlationID: String? = nil,
-        payPalContextID: String? = nil,
-        completion: @escaping (Error?) -> Void = { _ in }
-    ) {
-        DispatchQueue.main.async {
-            self.payPalContextID = payPalContextID
-            self.enqueueEvent(eventName, errorDescription: errorDescription, correlationID: correlationID)
-            self.flush(completion)
-        }
-    }
-
-    /// Executes API request to FPTI
-    func flush(_ completion: @escaping (Error?) -> Void = { _ in }) {
+        payPalContextID: String? = nil
+    ) async {
+        self.payPalContextID = payPalContextID
+        
+        let timestampInMilliseconds = UInt64(Date().timeIntervalSince1970 * 1000)
+        let event = FPTIBatchData.Event(
+            correlationID: correlationID,
+            errorDescription: errorDescription,
+            eventName: eventName,
+            timestamp: String(timestampInMilliseconds)
+        )
+                
         apiClient.fetchOrReturnRemoteConfiguration { configuration, error in
             guard let configuration, error == nil else {
-                if let error {
-                    completion(error)
```

Comment:
This completion was always ignored, so we removed it entirely

Reply:
Curious why this is happening, thoughts?

Reply:
`BTAPIClient` was always no-oping the result of this completion. I don't think we would ever want to return an error to the merchant and stop the flow if there is an issue with analytics since that's just for us, so this is safe for us to remove since we just discarded the result.